### DISCUSSION TITLE
missing publication information in pdf render API

### DISCUSSION THREAD

**Discussion by OliverStingl** [OP-TED/eForms-SDK#962] [Q&A]
After publishing, information under "Publication information" is still missing in the PDF rendered by API call to "https://viewer.ted.europa.eu/api/v2/notices/render", the heading itself is present.
In the PDF downloaded via direct links on ted.europa.eu ("https://ted.europa.eu/en/notice/[publication-id]/pdf") publication information is present:
![image](https://github.com/OP-TED/eForms-SDK/assets/52192933/05479f90-f9bd-4c87-a7c0-fd291eacb611)

POST-data for curl is as following (PHP Curl):
[
    'file' => base64_encode($xml),
    'language' => 'de',
    'format' => 'PDF',
    'textToHighlight' => 'string',
    'summary' => false,
];

**Comment by pdonohoe** [comment]
Dear @OliverStingl , I cannot replicate the error you see. I have called the live render API https://viewer.ted.europa.eu/api/v2/notices/render with the notice XML, and the returned PDF includes the publication information, see attached PDF. 
[2024-OJS077-00230964-live.pdf](https://github.com/OP-TED/eForms-SDK/files/15189905/2024-OJS077-00230964-live.pdf)

**Comment by OliverStingl** [comment]
We are sending the notice-xml to the render api and we still receive the PDF without any publication information. Is the notice XML altered on the TED-portal after publication? We're sending the exact XML we're sending for publishing.

**Comment by bertrand-lorentz** [comment]
The publication information (OJ S issue number, Notice publication number and publication date) is added in the notice XML as part of the publication process. You can see this information in the XML available on the TED website.

The service at "https://viewer.ted.europa.eu/api/v2/notices/render" only converts the provided XML notice to HTML or PDF, it does not fetch the publication information to add it to the output.

I guess @pdonohoe used the XML notice retrieved from the TED website, so the publication information was reflected in the PDF.

If you want to get the notice in PDF "as published", I would recommend to download it directly from the TED website.

(Moving this to a discussion, as it is not an issue in the SDK)

**Reply by OliverStingl** [reply]
Thanks! 
Is there any way to get a direct link to the xml via the publication ID like on the TED website, but for the preview environment?

**Reply by OliverStingl** [reply]
Also, i just checked the xml available on the TED websites. They have an entirely different structure that is not recognized by the render api at all, all response i get is a code 500.

**Reply by bertrand-lorentz** [reply]
From what I know:
1/ There is no TED website for the preview environment (see https://docs.ted.europa.eu/eforms-common/preview/index.html)
2/ The XML on the TED website should be the same except for the additions I indicated.

For further questions concerning the TED website, please contact their Helpdesk: https://ted.europa.eu/en/contact

**Reply by OliverStingl** [reply]
So there is no way to recieve the complete XML via the API? 
I couldn't find anything in the documentation.
